ls1intum/Artemis
# [Bugfix] Several fixes for Programming Exercises

Fixes #2763 
Fixes #2760 




### Checklist

### Motivation and Context


This PR fixes several minor bugs found in different testing sessions (see also issues attached).

### Description

- Show result string instead of Build Failed for manual programming assessments
- Extended query for locked submission 
- Add More Feedback Request label
- Remove buttons for tutors in course and exercise details view
- Load assessor eagerly

### Steps for Testing


1. Create a Programming exercise, participate, assess and make a complaint (Care, the assessor musn't be the student)
2. Accept the complaint. After that, the original assessor should be correctly displayed (See screenshot)
3. In another submission, ask for more feedback. When the assessor clicks on the button to give more feedback on the top right corner "You have to respond to the feedback request" should be displayed (See screenshot)

1. Navigate to the course detail page and programming-exercise detail page with an student account and make sure you cannot see tutor or instructor related buttons (See screenshot)


1. Lock 10 submissions and make sure you cannot lock another one. You can see your locks in the 'Assessments locked by you' in the assessment dashboard.
1. Participate in an programming exercise with manual assessment and make sure the build fails (e.g. remove a ; in the end of a line)
2. Assess the exercise. For the student, the scores should be displayed afterwards (See screenshot)

### Test Coverage



Nothing changed.

### Screenshots


Course Detail View:
![image](https://user-images.githubusercontent.com/41108487/106056936-80d7ab00-60ef-11eb-9457-984c82d498b2.png)

Exercise Detail View:
![image](https://user-images.githubusercontent.com/41108487/106056976-92b94e00-60ef-11eb-9f56-e45fe86ebb25.png)


Locked Submission View:
![image](https://user-images.githubusercontent.com/41108487/106057142-ceecae80-60ef-11eb-8aa3-b52be83f8bf7.png)

Locked Submission Limit:
![image](https://user-images.githubusercontent.com/41108487/106057214-ef1c6d80-60ef-11eb-8501-35ad1a36d2c0.png)


Build Failed result string:
![image](https://user-images.githubusercontent.com/41108487/106056627-2cccc680-60ef-11eb-816f-c5ff6195ff34.png)

After evaluating and accepting the complaint:

![image](https://user-images.githubusercontent.com/41108487/106060709-ab783280-60f4-11eb-85de-b7b6a84c8bf3.png)


More feedback request:
![image](https://user-images.githubusercontent.com/41108487/106055394-98ae2f80-60ed-11eb-8a6c-52dcfd4fd77f.png)
